### PR TITLE
JRPC - debug methods review

### LINKED ISSUE #1598: test jRPC debug methods against L1 and L2
We need to test the debug endpoints against both l1 and l2 networks in order to make sure they are working exactly the same way

### PR DESCRIPTION
**Pull request by tclemos**
Closes #1598.

### What does this PR do?

Add e2e tests to guarantee the jRPC debug methods are behaving on l2 exactly as it works on l1

### Reviewers

Main reviewers:

@ToniRamirezM 
@arnaubennassar

### CODE REVIEW

File: jsonrpc/endpoints_debug.go [hunk 1]
```diff
@@ -34,89 +37,188 @@ type StructLogRes struct {
...
 		}
 
-		resp := traceTransactionResponse{
-			Gas:         result.GasUsed,
-			Failed:      failed,
-			ReturnValue: common.Bytes2Hex(result.ReturnValue),
-			StructLogs:  structLogs,
+		op := structLog.Op
+		if op == "SHA3" {
+			op = "KECCAK256"
```
Comment (ToniRamirezM) [inline jsonrpc/endpoints_debug.go:176]: Why? Maybe adding a comment could help

File: state/state.go [hunk 2]
```diff
@@ -926,6 +942,17 @@ func (s *State) DebugTransaction(ctx context.Context, transactionHash common.Has
 	}
 	endTime := time.Now()
 
+	// //save process batch response file
```
Comment (ToniRamirezM) [inline state/state.go:945]: This can be deleted

**Review by ToniRamirezM — APPROVED** [review]

File: jsonrpc/endpoints_debug.go (hunk 1 above)
Comment (tclemos) [reply jsonrpc/endpoints_debug.go:176]: I just followed the values returned by Geth and converted the info returned by the executor to it. Maybe we could ask the executor team "why".

File: state/state.go (hunk 2 above)
Comment (tclemos) [reply state/state.go:945]: Would you mind if I keep it commented in the test? This helped a lot extracting data to provide to the executor and protocol team in order to understand what was not working.

File: jsonrpc/endpoints_debug.go (hunk 1 above)
Comment (tclemos) [reply jsonrpc/endpoints_debug.go:176]: our source: https://www.evm.codes/#20?fork=berlin
image

**Pull request merged**